### PR TITLE
Add webassembly arch 

### PR DESCRIPTION
**Pull request by SpikeI**
Add webassembly arch

### CODE REVIEW

**Comment by aquynh** [comment]
merged, thanks for this amazing work!

**Comment by aquynh** [comment]
look forward to having Python binding support for Web Assembly next!

File: CMakeLists.txt [hunk 1]
```diff
@@ -119,7 +121,7 @@ set(HEADERS_COMMON
 set(TEST_SOURCES test_basic.c test_detail.c test_skipdata.c test_iter.c)
 
 ## architecture support
-if (CAPSTONE_ARM_SUPPORT)
+if (NOT CAPSTONE_X86_ONLY AND CAPSTONE_ARM_SUPPORT)
```
Comment (radare) [inline CMakeLists.txt]: all this seems unrelated to the pr

File: SStream.c [hunk 2]
```diff
@@ -66,8 +66,7 @@ void printInt64Bang(SStream *O, int64_t val)
 				SStream_concat(O, "#-0x%"PRIx64, (uint64_t)val);
 			else
 				SStream_concat(O, "#-0x%"PRIx64, (uint64_t)-val);
-		}
-		else
+		} else
```
Comment (radare) [inline SStream.c]: changing coding style? is the capstone coding style documented anywhere?

File: arch/WASM/WASMInstPrinter.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+void WASM_printInst(MCInst *MI, struct SStream *O, void *PrinterInfo)
+{
+	int i;
+
+	SStream_concat(O, WASM_insn_name((csh)MI->csh, MI->Opcode));
+
+	switch (MI->wasm_data.type) {
+		default:
```
Comment (radare) [inline arch/WASM/WASMInstPrinter.c]: Is placing the default case at the beginning also part of the capstone coding style? didn't saw that before

File: arch/WASM/WASMMapping.c [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+
+const char *WASM_insn_name(csh handle, unsigned int id)
+{
+#ifndef CAPSTONE_DIET
+	if (id >= ARR_SIZE(insn_name_maps))
+		return NULL;
+	else
+		return insn_name_maps[id].name;
+#else
+	return NULL;
```
Comment (radare) [inline arch/WASM/WASMMapping.c]: in diet mode wasm disassembler not work

File: cs.c [hunk 5]
```diff
@@ -1381,11 +1403,18 @@ int CAPSTONE_API cs_op_count(csh ud, const cs_insn *insn, unsigned int op_type)
...
 					count++;
 #endif
+			break;
 		case CS_ARCH_MOS65XX:
 			for (i = 0; i < insn->detail->m680x.op_count; i++)
 				if (insn->detail->m680x.operands[i].type == (m680x_op_type)op_type)
 					count++;
 			break;
+		case CS_ARCH_WASM:
+#if 0
```
Comment (radare) [inline cs.c]: dead code

File: cstool/cstool.c [hunk 6]
```diff
@@ -134,8 +136,8 @@ static uint8_t *preprocess(char *code, size_t *size)
 
 static void usage(char *prog)
 {
-	printf("cstool for Capstone Disassembler, v%u.%u.%u\n\n", CS_VERSION_MAJOR, CS_VERSION_MINOR, CS_VERSION_EXTRA);
-	printf("Syntax: %s [-d|-u|-s|-v] <arch+mode> <assembly-hexstring> [start-address-in-hex-format]\n", prog);
+	printf("Cstool for Capstone Disassembler Engine v%u.%u.%u\n\n", CS_VERSION_MAJOR, CS_VERSION_MINOR, CS_VERSION_EXTRA);
+	printf("Syntax: %s [-u|-d|-s|-v] <arch+mode> <assembly-hexstring> [start-address-in-hex-format]\n", prog);
```
Comment (radare) [inline cstool/cstool.c]: flags sohuld be sorted alphabetically, and this change is unrelated to the pr

Comment (aquynh) [reply cstool/cstool.c]: i will fix the flag issue

File: CMakeLists.txt (hunk 1 above)
Comment (aquynh) [reply CMakeLists.txt]: he ported his pull req from older branch, so mixed up with newer code. this is from old unrelated commit, but still shows up here

File: SStream.c (hunk 2 above)
Comment (aquynh) [reply SStream.c]: no, we always use Linux kernel style. i just updated HACK.TXT with this info: https://github.com/aquynh/capstone/commit/864595c3559a27fbcddc32b15f59db9e74342d2e

File: arch/WASM/WASMInstPrinter.c (hunk 3 above)
Comment (aquynh) [reply arch/WASM/WASMInstPrinter.c]: i am not sure if Linux kernel coding style follows this, but personally i find that having default case at the top is easier to track.

File: arch/WASM/WASMMapping.c (hunk 4 above)
Comment (aquynh) [reply arch/WASM/WASMMapping.c]: @SpikeI can you take a look at Diet mode?

File: bindings/python/capstone/__init__.py [hunk 7]
```diff
@@ -129,8 +129,8 @@
 # Capstone C interface
 
 # API version
-CS_API_MAJOR = 4
-CS_API_MINOR = 1
+CS_API_MAJOR = 5
```
[the remark this answers is not in the document]
Comment (aquynh) [reply bindings/python/capstone/__init__.py]: this is the same issue: older commits shows up here.

File: cs.c (hunk 5 above)
Comment (aquynh) [reply cs.c]: fixed with https://github.com/aquynh/capstone/commit/ee8bcfd1e717c1eaa096e40fb765248b9b0315cd

File: cstool/cstool.c (hunk 6 above)
Comment (aquynh) [reply cstool/cstool.c]: see https://github.com/aquynh/capstone/commit/5ae590943a7356c460d7616615397af502ebf201
